packages/cactus-plugin-keychain-azure-kv/src/main/typescript/plugin-keychain-azure-kv-remote-adapter.ts

```diff
@@ -0,0 +1,156 @@
+    Checks.truthy(
+      opts.backend instanceof DefaultApi,
+      `${fnTag} opts.backend instanceof DefaultApi`,
+    );
+
+    const level = this.opts.logLevel || "INFO";
+    const label = this.className;
+    this.log = LoggerProvider.getOrCreate({ level, label });
+
+    this.instanceId = this.opts.instanceId;
+    this.backend = opts.backend;
+    this.keychainId = opts.keychainId;
+
+    this.log.info(`Created ${this.className}. KeychainID=${opts.keychainId}`);
+  }
+
+  /**
+   * Dummy implementation that wires no web services on the host API server
+   * because there is no need. All the functionality is implemented somewhere
+   * else on a host that's accessible through the network to this object
+   * (because this class is a remote adapter not an actual plugin impl.).
+   *
+   * @param _expressApp
+   */
+  public async getOrCreateWebServices(): Promise<IWebServiceEndpoint[]> {
+    return [];
+  }
+
+  /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
+  public registerWebServices(app: Express): Promise<IWebServiceEndpoint[]> {
+    return this.getOrCreateWebServices();
+  }
+
+  public getHttpServer(): Optional<Server | SecureServer> {
+    return Optional.empty();
+  }
+
+  public async shutdown(): Promise<void> {
+    return;
+  }
+
+  public rotateEncryptionKeys(): Promise<void> {
+    throw new Error("Method not implemented.");
+  }
+
+  public getEncryptionAlgorithm(): string {
+    throw new Error("Method not implemented.");
+  }
+
+  public getKeychainId(): string {
+    return this.keychainId;
+  }
+
+  public async has(key: string): Promise<boolean> {
+    try {
+      await this.backend.getKeychainEntry({ key });
+      return true;
+    } catch (ex) {
+      // FIXME check for errors being thrown due to something other than
+      // the key not being present...
+      return false;
+    }
+  }
+
+  public async get<T>(key: string): Promise<T> {
+    const { data } = await this.backend.getKeychainEntry({ key });
+    // FIXME what to do here? Does it make any sense to have the get() method
+    // of the keychain be generically parameterizable when we know we can only
+    // return a string anyway?
+    return data.value as any;
+  }
+
+  public async set<T>(key: string, value: T): Promise<void> {
+    // FIXME Does it make any sense to have the set() method be generic?
+    await this.backend.setKeychainEntry({ key, value: value as any });
+  }
+
+  public async delete(key: string): Promise<void> {
+    // FIXME Pretty sure azure keyvault can do delete so we don't have to hack it like this
+    // but it cannot be done in this code until the rust code has been updated
+    // to have that endpoint as well...
+    await this.backend.setKeychainEntry({ key, value: "" });
+  }
+
+  public getInstanceId(): string {
+    return this.instanceId;
+  }
+
+  public getPackageName(): string {
+    return `@hyperledger/cactus-plugin-keychain-azure-kv`;
+  }
+
+  public getAspect(): PluginAspect {
```

Comment:
@jagpreetsinghsasan This method can be deleted IIRC. Once you rebase onto main the plugin aspect should cease to exist.